appinventor/components/src/com/google/appinventor/components/runtime/Ev3Motor.java

```diff
@@ -0,0 +1,671 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2011-2012 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.components.runtime;
+
+import com.google.appinventor.components.annotations.DesignerComponent;
+import com.google.appinventor.components.annotations.DesignerProperty;
+import com.google.appinventor.components.annotations.PropertyCategory;
+import com.google.appinventor.components.annotations.SimpleFunction;
+import com.google.appinventor.components.annotations.SimpleEvent;
+import com.google.appinventor.components.annotations.SimpleObject;
+import com.google.appinventor.components.annotations.SimpleProperty;
+import com.google.appinventor.components.common.ComponentCategory;
+import com.google.appinventor.components.common.PropertyTypeConstants;
+import com.google.appinventor.components.common.YaVersion;
+import com.google.appinventor.components.runtime.util.Ev3Constants;
+import com.google.appinventor.components.runtime.util.Ev3BinaryParser;
+import com.google.appinventor.components.runtime.util.ErrorMessages;
+
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.List;
+
+import android.os.Handler;
+/**
+ * A component that provides both high- and low-level interfaces to
+ * control the motors on LEGO MINDSTORMS EV3.
+ *
+ * @author jerry73204@gmail.com (jerry73204)
+ * @author spaded06543@gmail.com (Alvin Chang)
+ */
+@DesignerComponent(version = YaVersion.EV3_MOTOR_COMPONENT_VERSION,
+                   description = "A component that provides both high- and low-level interfaces to a LEGO MINDSTORMS EV3 " +
+                                 "robot, with functions that can control the motors.",
+                   category = ComponentCategory.LEGOMINDSTORMS,
+                   nonVisible = true,
+                   iconName = "images/legoMindstormsEv3.png")
+@SimpleObject
+public class Ev3Motor extends LegoMindstormsEv3Base {
```

Comment:
Since Ev3Motor can control more than one motor, maybe it should be named Ev3Motors.